### ISSUE TITLE
Implement lcar table parsing

### ISSUE THREAD

**Issue by KrasnayaPloshchad** [harfbuzz/harfbuzz#907]
This table is used to indicate ligature carets in AAT fonts.
See: https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6lcar.html

**Comment by ebraminio** [comment]
(just a personal note)

List of Apple fonts on macOS 10.13.3 with `lcar` table:

```
/Library/Fonts/Apple Chancery.ttf
/Library/Fonts/Hoefler Text.ttc
```

**Comment by ebraminio** [comment]
@behdad Only two fonts have it, maybe it is a sign that this is being replaced with GDEF's or the reverse way? The number is not changed between Sierra and High Sierra.

**Comment by ebraminio** [comment]
Needed code for this,

```C
struct lcar
{
  static const hb_tag_t tableTag = HB_AAT_TAG_lcar;

  inline bool sanitize (hb_sanitize_context_t *c) const
  {
    TRACE_SANITIZE (this);
    return_trace (c->check_struct (this) && lookup.sanitize (c));
  }

  protected:
  FixedVersion<>version;	/* Version number of the ligature caret table */
  HBUINT16	format;		/* Format of the ligature caret table. */
  Lookup<OffsetTo<ArrayOf<HBINT16> > >
		lookup;		/* data Lookup table associating glyphs */
  public:
  DEFINE_SIZE_MIN (8);
};
```

But `Lookup<OffsetTo<ArrayOf<HBINT16> > >` won't work so it needs runtime check and in order to make that happen and provide it seamlessly with OT's API this needs control points data from `glyf` table (format 1) which isn't there also and too much work just for supporting just two fonts and for an API (`hb_ot_layout_get_ligature_carets`) not used on Firefox and Chromium even.

**Comment by ebraminio** [comment]
A possible fix is https://github.com/harfbuzz/harfbuzz/pull/1358

**Comment by behdad** [comment]
> But `Lookup<OffsetTo<ArrayOf<HBINT16> > >` won't work as is and in order to make that happen and provide it seamlessly with OT's API this needs control points data from `glyf` table (format 1) which isn't there also and too much work just for supporting just two fonts and for an API (`hb_ot_layout_get_ligature_carets`) not used on Firefox and Chromium even.

I don't understand what you're saying.

**Comment by behdad** [comment]
Why did you close?  Is https://github.com/harfbuzz/harfbuzz/pull/1358 still relevant?

**Comment by ebraminio** [comment]
Oh it was closed when I didn't know there is someway to query glyf table already implemented and thought due to low number of fonts I thought let's give this up and you agreed also due to low number here https://github.com/harfbuzz/harfbuzz/pull/946 last comment

**Comment by ebraminio** [comment]
Fixed on #1358 :)
